cub/cub/device/dispatch/dispatch_find.cuh

```diff
@@ -0,0 +1,260 @@
+      if (cudaSuccess != error)
+      {
+        break;
+      }
+
+      int max_blocks = find_if_sm_occupancy * sm_count; // no * CUB_SUBSCRIPTION_FACTOR(0) because max_blocks gets too
+                                                        // big
+      const int findif_grid_size = ::cuda::std::min(num_tiles, max_blocks);
+
+      // Temporary storage allocation requirements
+      void* allocations[1]       = {};
+      size_t allocation_sizes[1] = {sizeof(int)};
+      // Alias the temporary allocations from the single storage blob (or
+      // compute the necessary size of the blob)
+      error = CubDebug(detail::AliasTemporaries(d_temp_storage, temp_storage_bytes, allocations, allocation_sizes));
+      if (cudaSuccess != error)
+      {
+        break;
+      }
+
+      OffsetT* value_temp_storage = static_cast<OffsetT*>(allocations[0]);
+
+      if (d_temp_storage == nullptr)
+      {
+        // Return if the caller is simply requesting the size of the storage
+        // allocation
+        return cudaSuccess;
+      }
+
+      // use d_temp_storage as the intermediate device result
+      // to read and write from. Then store the final result in the output iterator.
+
+      THRUST_NS_QUALIFIER::cuda_cub::detail::triple_chevron(1, 1, 0, stream)
+        .doit(detail::cuda_mem_set_async_dtemp_storage<OffsetT, OffsetT>, value_temp_storage, num_items);
+
+      // Invoke FindIfKernel
+      THRUST_NS_QUALIFIER::cuda_cub::detail::triple_chevron(
+        findif_grid_size, ActivePolicyT::FindPolicy::BLOCK_THREADS, 0, stream)
+        .doit(find_kernel, d_in, d_out, num_items, value_temp_storage, scan_op);
+
+      THRUST_NS_QUALIFIER::cuda_cub::detail::triple_chevron(1, 1, 0, stream)
+        .doit(
+          detail::write_final_result_in_output_iterator_already<OffsetT, OutputIteratorT>, value_temp_storage, d_out);
+
+      // Check for failure to launch
+      error = CubDebug(cudaPeekAtLastError());
+      if (cudaSuccess != error)
+      {
+        break;
+      }
+
+      // Sync the stream if specified to flush runtime errors
+      error = CubDebug(detail::DebugSyncStream(stream));
+      if (cudaSuccess != error)
+      {
+        break;
+      }
+
+    } while (0);
+    return error;
+  }
+
+  template <typename ActivePolicyT>
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE cudaError_t Invoke()
+  {
+    using MaxPolicyT = typename SelectedPolicy::MaxPolicy;
+    return Invoke<ActivePolicyT>(
+      detail::DeviceFindKernel<MaxPolicyT, InputIteratorT, OutputIteratorT, OffsetT, ScanOpT>); // include the
+                                                                                                // surrounding two init
+                                                                                                // and write back
+                                                                                                // kernels here.
```

Comment:
Question: is this comment still applicable?

Reply:
yes look: https://github.com/NVIDIA/cccl/blob/7619f6947ef44fbd2175915bba77e5ae219322e8/cub/cub/device/dispatch/dispatch_find.cuh#L183-L193